vignettes/netCDF_Functionality.Rmd

```diff
@@ -0,0 +1,400 @@
+necessary for interpreting the data.
+
+Different attribute types include variable attributes, time attributes, depth
+attributes, and global attributes. The format of attributes is a named list. Each attribute list
+has a bare minimum of required fields. Endless other fields are optional and can
+be added by the user as they see fit.
+
+### Variable Attributes
+
+A named list describing the variable data. The *name* and *units* fields are required.
+
+```{r, VarAtt, results = 'hide'}
+# single variable
+var_attributes <- list(
+  name = 'WatYrWDD',
+  long_name = 'Water Year Wet Degree Days',
+  units = 'number of days'
+)
+
+# multiple variables
+var_attributes2 <- list(
+    name = paste0('VarDD_', 1:10),
+    units = rep('number of days', 10)
+)
+
+```
+
+### Time Attributes
+
+A named list describing the time dimension. The *name*, *units*, *calender*, and *vals*
+fields are required here.
+
+A common unit for netCDF is days since 1900. A full list of accepted time units can
+be found [here](https://ncics.org/portfolio/other-resources/udunits2/).
+
+```{r, TimeAtt, results = 'hide'}
+time_attributes <- list(
+  name = 'time',
+  units = 'days since 1900-01-01',
+  calendar = 'standard',
+  unlim = TRUE,
+  vals = time_midpoints # mid point of year
+  )
+```
+
+### Vertical (depth / Z) Attributes
+
+A named list describing the vertical (depth / Z) dimension. The *name*, *units*,
+and *vals* fields are required.
+
+```{r, VertAtt, results = 'hide'}
+vertical_attributes <- list(
+  name = 'depth',
+  units = 'cm',
+  positive = 'down', # depth is down, height is up
+  vals = depths_max
+)
+```
+
+### Global Attributes
+
+A named list describing the global attributes of the netCDF. This is where you describe
+things like how the data was made, where the data was made, who the data was made by,
+what climate models were used to create the data, etc.
+
+The *title*, *institution*, *description*, *source_id*, *realm*, *product*, and
+*contact* fields are required.
+
+```{r, GlobalAtt, results = 'hide'}
+global_attributes <- list(
+    title = "Project Name Annual Wet Degree Days",
+    institution = 'Southwest Biological Science Center, U.S. Geological Survey',
+    description = 'methods on how this data was made',
+    source = paste(
+      "SOILWAT2 (v4.2.0);",
+      "rSOILWAT2 (v2.3.2);",
+      "rSFSW2 (v3.1.2)"
+    ),
+    source_id = "SOILWAT2",
+    realm = "land",
+    parent_mip_era = "CMIP5",
+    parent_experiment_id = "RCP45",
+    parent_source = "CanESM2",
+    parent_variant_label = "r1i1p1f1",
+    product = "model-output",
+    projection = 'Geographic',
+    grid = 'WGS84',
+    grid_label = "gn",
+    nominal_resolution = "10 km", # \code{\link{populate_netcdf_from_array}}
+    further_info_url = "https://github.com/DrylandEcology/",
+    contact = "you@email.com"
+    )
+```
+
+#### Nominal resolution
+
+If the data is calculated and reported on a grid, CF convention calls for the
```

Comment:
It's not in the CF-1.8 convention; I'm pretty sure that "nominal resolution" is a CMIP6 standard attribute. See "Appendix 2: Algorithms for Defining the “nominal_resolution” Attribute" (https://docs.google.com/document/d/1h0r8RZr_f3-8egBMMh7aqLwy3snpD6_MrDz1q8n5XUk)